### PR TITLE
Add user auth

### PR DESCRIPTION
**Pull request by savsa**
- Change what's displayed on the page based on the auth status
- Have comments be displayed with the newest one on top
- Add navbar to show log in button and the user's email address
- Hide the comment box if the user is not logged in

Screenshots:
https://screenshot.googleplex.com/ZDUffxfCvZO.png
https://screenshot.googleplex.com/q7YCJZDzjGC.png

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -38,6 +38,11 @@
       <artifactId>appengine-api-1.0-sdk</artifactId>
       <version>1.9.59</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-api-1.0-sdk</artifactId>
+      <version>1.9.59</version>
+    </dependency>
```
Comment (triplehair) [inline portfolio/pom.xml:45]: Are these already added above?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -26,17 +26,37 @@ document.querySelector('.comment-form').addEventListener('submit', async functio
...
+    loginLink.className = 'login-link';
+    loginLink.innerText = 'Log in';
+    nav.append(loginLink);
+
+    document.querySelector('.comment-form').style.display = 'none';
+    return;
+  }
+
+  // The user is logged in.
+  let p = document.createElement('p');
```
Comment (triplehair) [inline portfolio/src/main/webapp/script.js:51]: Change p to some meaningful variable name

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -26,17 +26,37 @@ document.querySelector('.comment-form').addEventListener('submit', async functio
...
   }
-  appendComment(document.querySelector('.comment-box').value);
+  rerequestComments();
 });
 
+// Load the dynamic parts of the page like comments and the authentication status.
+async function load() {
+  const response = await fetch(`/comment?num=5`);
+  const json = await response.json();
+  appendComments(json.comments);
```
Comment (triplehair) [inline portfolio/src/main/webapp/script.js:36]: Call getComments(5) instead?

Comment (savsa) [reply portfolio/src/main/webapp/script.js:36]: I use the `json` variable below this to get the user's email.

**Pull request merged**